packages/browser/src/lib/client-hints/index.ts

```diff
@@ -0,0 +1,25 @@
+export async function clientHints(
+  hints?: string[]
+): Promise<UADataValues | undefined> {
+  const userAgentData = ((): NavigatorUAData | undefined =>
+    (navigator as any).userAgentData)()
+
+  if (!userAgentData) return
+
+  let userAgentDataContext: UADataValues = {
+    brands: userAgentData.brands,
+    mobile: userAgentData.mobile,
+    platform: userAgentData.platform,
+  }
+
+  if (hints) {
+    try {
+      // this also includes the low entropy values, so we can overwrite here
+      userAgentDataContext = await userAgentData.getHighEntropyValues(hints)
+    } catch (_) {
+      return userAgentDataContext
+    }
+  }
+
+  return userAgentDataContext
+}
```

Comment:
Nit: imo a shorter and a bit more 'functional' way to write this would be: 

```ts

export async function clientHints(...) {
  if (!(navigator as any).userAgentData) return undefined;
  
  const userAgentDataContext: UADataValues = {
      brands: userAgentData.brands,
      mobile: userAgentData.mobile,
      platform: userAgentData.platform,
    }
    
  if (!hints) return userAgentDataContext
  return userAgentData.getHighEntropyValues(hints).catch(() => userAgentDataContext)
}
```